code/datums/diseases/advance/symptoms/goat.dm

```diff
@@ -0,0 +1,137 @@
+		var/mob/living/carbon/human/H = A.affected_mob	
+		ADD_TRAIT(H, TRAIT_TRASH_EATER, type)
+
+	if((A.stage >= 5) && (has_headbutt == TRUE)) 				//Checks for existing headbutt ability before granting it.
+		var/mob/living/carbon/C = A.affected_mob
+		if(locate(/obj/effect/proc_holder/spell/aimed/headbutt) in C.mob_spell_list) //Check for existing headbutt.
+			return
+		var/obj/effect/proc_holder/spell/aimed/headbutt/headbuttgrant = new/obj/effect/proc_holder/spell/aimed/headbutt
+		M.AddSpell(headbuttgrant)
+		has_headbutt = FALSE
+		
+/datum/symptom/goat/End(datum/disease/advance/A)
+	. = ..()
+	var/mob/living/M = A.affected_mob
+	M.remove_movespeed_modifier(type)
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = A.affected_mob	
+		REMOVE_TRAIT(H, TRAIT_TRASH_EATER, type)
+	M.RemoveSpell(/obj/effect/proc_holder/spell/aimed/headbutt)
+	UnregisterSignal(M, COMSIG_MOB_SAY)
+
+/datum/symptom/goat/proc/handle_speech(datum/source, list/speech_args) //Creates a "baa" effect on the infected's speech
+	SIGNAL_HANDLER
+
+	var/message = speech_args[SPEECH_MESSAGE]
+	var/list/split_message = splittext(message, "")
+	for (var/i in 1 to length(split_message))
+		if(findtext(split_message[i], "a"))
+			if(prob(25))
+				split_message[i] = pick("a-a-a", "a-a")
+	message = jointext(split_message, "")
+	speech_args[SPEECH_MESSAGE] = message
+	
+/obj/effect/proc_holder/spell/aimed/headbutt
+	name = "Headbutt"
+	desc = "Causes you to charge forward towards a nearby target"
+	charge_type = "recharge"
+	charge_max	= 150
+	cooldown_min = 150
+	clothes_req = FALSE
+	projectile_type = null
+	action_icon = 'icons/mob/actions/actions_viro.dmi'
+	action_background_icon_state = "bg_viro"
+	base_icon_state = "viro_goat0"
+	action_icon_state = "viro_goat1"
+	active_msg = "You ready yourself to headbutt!"
+	deactive_msg = "You relax your muscles."
+	
+/obj/effect/proc_holder/spell/aimed/headbutt/update_icon()
+	if(!action)
+		return
+	if(active)
+		action.button_icon_state = base_icon_state
+	else
+		action.button_icon_state = action_icon_state
+	action.UpdateButtonIcon()
+	return
+
+/obj/effect/proc_holder/spell/aimed/headbutt/cast(list/targets, mob/user = usr)
+	var/target = get_turf(targets[1])
+	var/mob/living/M = user
+	. = ..()
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = user
+		if(H.legcuffed) 					//No headbutts while your LEGS are cuffed!
+			user.visible_message("<span class='warning'>[user] trips while trying to charge!</span>")
+			M.Paralyze(20)
+			return
+	user.throwforce = 25 //Just enough to very slowly break a door, window or crate. 25 hits on a crate.
+	user.emote("scream")
+	user.visible_message("<span class='warning'>[user] charges forward!</span>")
+	user.throw_at(target, 4, 5)
+	M.Paralyze(20) //You're stunned, hit or miss
+	sleep(20)
+	user.throwforce = 10
+	M.adjustOrganLoss(ORGAN_SLOT_BRAIN, 2) //Very minor brain damage since you're hitting your head against things.

```

Comment:
You probably dont need this sleep here, throw_at will handle the throwforce before it gets set back to 10

Reply:
Will it?
Wasn't sure if a target would get their throwforce set back to 10 before actually hitting a target

Reply:
It goes through the whole proc before going to the next, if throw_at also handles the hit or has another proc in it that does, you won't need to sleep.

Reply:
Alright then, you're fine to keep it.

Reply:
With sleep its not really performance, it stops the proc from being called until its done, but that would only be a thing with admemes (and they can already throw you at things)